udata_front/theme/gouvfr/assets/js/pages/ReusePublishingForm/ReusePublishingForm.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+  <div class="fr-container" ref="containerRef">
+    <Step1DescribeReuse
+      v-if="currentStep === 0 && me"
+      :reuse="reuse"
+      :steps="steps"
+      :errors="errors"
+      :user="me"
+      @next="createReuseAndMoveToNextStep"
+    />
+    <Step2AddDatasets
+      v-else-if="currentStep === 1"
+      :steps="steps"
+      :errors="errors"
+      @next="updateDatasetsAndMoveToNextStep"
+    />
+    <Step3CompleteThePublication
+      v-else-if="currentStep === 2"
+      :steps="steps"
+      :feedbackUrl="publishing_form_feedback_url"
+      :originalReuse="savedReuse"
+      @update="updateReuseData"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onMounted, ref, toValue } from 'vue';
+import { useI18n } from 'vue-i18n';
+import type { Dataset, User } from '@datagouv/components';
+import Step1DescribeReuse from './Step1DescribeReuse.vue';
+import Step2AddDatasets from './Step2AddDatasets.vue';
+import Step3CompleteThePublication from './Step3CompleteThePublication.vue';
+import { publishing_form_feedback_url } from '../../config';
+import { createReuse, updateReuse, uploadLogo } from '../../api/reuses';
+import type { NewReuse, Reuse } from '../../types';
+import { fetchMe } from '../../api/me';
+import { auth } from '../../plugins/auth';
+  
+const { t } = useI18n();
+
+const steps = [t("Describe your reuse"), t("Link datasets"), t("Complete your publishing")];
+
+const currentStep = ref<number>(0);
+
+const containerRef = ref<HTMLDivElement | null>(null);
+
+const me = ref<User | null>(null);
+
+const user = auth();
+
+const reuse = ref<NewReuse>({
+  title: "",
+  description: "",
+  tags: null,
+  url: "",
+  datasets: [],
+  topic: "",
+  type: "",
+  organization: null,
+  owner: user.id,
+});
+
+const savedReuse = ref<Reuse>({
+  id: "",
+  page: "",
+  private: false,
+  deleted: false,
+  image: "",
+  image_thumbnail: "",
+  slug: "",
+  last_update: "",
+  title: "",
+  description: "",
+  tags: null,
+  url: "",
+  datasets: [],
+  topic: "",
+  type: "",
+  organization: null,
+  owner: user,
+});
+
+const errors = ref<Array<string>>([]);
+
+const moveToStep = (step: number, saveToHistory = true) => {
+  if(containerRef.value) {
+    containerRef.value.scrollIntoView({
+      behavior: "smooth"
+    });
+  }
+  if(step !== null) {
+    currentStep.value = step;
+    if(saveToHistory) {
+      let url = new URL(window.location.href);
+      const targetHash = `#/step-${step + 1}`;
+      url.hash = targetHash;
+      window.history.pushState(null, "", url);
+    }
+  }
+};
+
+async function createReuseAndMoveToNextStep(newReuse: NewReuse, file: File) {
+  errors.value = [];
+  let moveToNextStep = false;
+  try {
+    savedReuse.value = await createReuse(newReuse);
+    moveToNextStep = true;
+  } catch (e) {
+    errors.value.push(e.message);
+  }
+  try {
+    const resp = await uploadLogo(savedReuse.value.id, file);
```

Comment:
This produces an error to me. The request sent is `http://dev.local:7000/api/1/reuses//image/` without the id

Reply:
Note : I have the same 405 error even when the id is set e.g. `http://dev.local:7000/api/1/reuses/66aca33d74f4fe4b00bea1a7/image/`

Reply:
The valid url is without a trailing `/`

Reply:
This isn't resolved. If the reuse creation fail with a 400 (like a link already registered), the image shouldn't be sent and a relevant error message should be displayed.

Reply:
Note : the error message shown is :
```
Request failed with status code 400
Failed to upload logo, you can upload it again in your management panel
```